File: Add Code Here/PYTHON/fibonacci.py

def fib(n: int) -> list[int]:
    fibonacci_sequence = []
    a, b = 0, 1

    while len(fibonacci_sequence) < n:
        fibonacci_sequence.append(a)
        a, b = b, a + b

    return fibonacci_sequence

def recursive_fib(n: int) -> list[int]:
    if n <= 0:
        return []
    if n == 1:
        return [0]
    if n == 2:
        return [0, 1]
    fib_sequence = recursive_fib(n - 1)
    fib_sequence.append(fib_sequence[-1] + fib_sequence[-2])
    return fib_sequence
    
def fib_generator():
    a, b = 0, 1
    while True:
        yield a
        a, b = b, a + b

    
if __name__ == "__main__":
    fib_gen = fib_generator()

    n = input("Number of elements of the sequence:\n")
    n = int(n)

    print(fib(n))
    print(recursive_fib(n))

    for _ in range(n):
        print(next(fib_gen))

File: Add Code Here/C++/fibonacci.c
#include<stdio.h>

int fib(int n) {
  static int fib_table[1005];
  if (n < 0) {
    return -1;
  } else if (n <= 1) {
    return n;
  } else if (fib_table[n] != -1) {
    return fib_table[n];
  } else {
    fib_table[n] = fib(n - 1) + fib(n - 2);
    return fib_table[n];
  }
}

void main()
{
	int i,j;
	printf("Enter number of elements in sequence:\n");
	scanf("%d",&i);
	for(j=0;j<i;j++)
	{
		printf("%d ",fib(j));
	}
	
}
